### ISSUE TITLE
Unable to update pip dependencies (versions incompatibility)

### ISSUE THREAD

**Issue by Pitel** [taigaio/taiga-back#1112]
**Do you want to request a *feature* or report a *bug*?**
*bug*

**What is the current behavior?**
```
(taiga) taiga@taiga:~/taiga-back$ pip install --upgrade -r requirements.txt
Collecting git+git://github.com/Python-Markdown/markdown.git@a4d4b61b5ce4a7dd96ddc13c66b8716d3ad8eb51 (from -r requirements.txt (line 4))
  Cloning git://github.com/Python-Markdown/markdown.git (to revision a4d4b61b5ce4a7dd96ddc13c66b8716d3ad8eb51) to /tmp/pip-req-build-8ht23ead
Collecting git+https://github.com/Xof/django-pglocks.git (from -r requirements.txt (line 25))
  Cloning https://github.com/Xof/django-pglocks.git to /tmp/pip-req-build-nzmd0hna
Requirement already up-to-date: CairoSVG==2.0.3 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 2)) (2.0.3)
Requirement already up-to-date: Django==1.11.2 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 3)) (1.11.2)
Requirement already up-to-date: Pillow==4.1.1 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 5)) (4.1.1)
Requirement already up-to-date: PyJWT==1.5.0 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 6)) (1.5.0)
Requirement already up-to-date: Unidecode==0.4.20 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 7)) (0.4.20)
Requirement already up-to-date: amqp==2.1.4 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 8)) (2.1.4)
Requirement already up-to-date: asana==0.6.5 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 9)) (0.6.5)
Requirement already up-to-date: bleach==2.0.0 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 10)) (2.0.0)
Requirement already up-to-date: celery==4.0.2 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 11)) (4.0.2)
Requirement already up-to-date: kombu==4.0.2 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 12)) (4.0.2)
Requirement already up-to-date: cryptography==1.9 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 13)) (1.9)
Requirement already up-to-date: cssutils==1.0.2 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 14)) (1.0.2)
Requirement already up-to-date: diff-match-patch==20121119 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 15)) (20121119)
Requirement already up-to-date: django-ipware==1.1.6 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 16)) (1.1.6)
Requirement already up-to-date: django-jinja==2.3.1 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 17)) (2.3.1)
Requirement already up-to-date: django-picklefield==0.3.2 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 18)) (0.3.2)
Requirement already up-to-date: django-sampledatahelper==0.4.1 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 19)) (0.4.1)
Requirement already up-to-date: django-sites==0.9 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 20)) (0.9)
Requirement already up-to-date: django-sr==0.0.4 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 21)) (0.0.4)
Requirement already up-to-date: djmail==1.0.1 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 22)) (1.0.1)
Requirement already up-to-date: easy-thumbnails==2.4.1 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 23)) (2.4.1)
Requirement already up-to-date: fn==0.4.3 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 24)) (0.4.3)
Requirement already up-to-date: gunicorn==19.7.1 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 26)) (19.7.1)
Requirement already up-to-date: jinja2==2.9.6 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 27)) (2.9.6)
Requirement already up-to-date: lxml==3.8.0 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 28)) (3.8.0)
Requirement already up-to-date: netaddr==0.7.19 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 29)) (0.7.19)
Requirement already up-to-date: premailer==3.0.1 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 30)) (3.0.1)
Requirement already up-to-date: psd-tools==1.4 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 31)) (1.4)
Requirement already up-to-date: psycopg2==2.7.1 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 32)) (2.7.1)
Requirement already up-to-date: pygments==2.2.0 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 33)) (2.2.0)
Requirement already up-to-date: pyjwkest==1.3.2 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 34)) (1.3.2)
Requirement already up-to-date: python-dateutil==2.6.0 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 35)) (2.6.0)
Requirement already up-to-date: python-magic==0.4.13 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 36)) (0.4.13)
Requirement already up-to-date: pytz==2017.2 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 37)) (2017.2)
Requirement already up-to-date: raven==6.1.0 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 38)) (6.1.0)
Requirement already up-to-date: redis==2.10.5 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 39)) (2.10.5)
Requirement already up-to-date: requests-oauthlib==0.8.0 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 40)) (0.8.0)
Requirement already up-to-date: requests==2.17.3 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 41)) (2.17.3)
Requirement already up-to-date: serpy==0.1.1 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 42)) (0.1.1)
Requirement already up-to-date: six==1.10.0 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 43)) (1.10.0)
Requirement already up-to-date: webcolors==1.7 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from -r requirements.txt (line 44)) (1.7)
Requirement not upgraded as not directly required: tinycss in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from CairoSVG==2.0.3->-r requirements.txt (line 2)) (0.4)
Requirement not upgraded as not directly required: cssselect in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from CairoSVG==2.0.3->-r requirements.txt (line 2)) (1.0.3)
Requirement not upgraded as not directly required: cairocffi in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from CairoSVG==2.0.3->-r requirements.txt (line 2)) (0.8.0)
Requirement not upgraded as not directly required: olefile in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from Pillow==4.1.1->-r requirements.txt (line 5)) (0.45.1)
Requirement not upgraded as not directly required: vine>=1.1.3 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from amqp==2.1.4->-r requirements.txt (line 8)) (1.1.4)
Requirement not upgraded as not directly required: html5lib>=0.99999999 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from bleach==2.0.0->-r requirements.txt (line 10)) (1.0.1)
Requirement not upgraded as not directly required: billiard<3.6.0,>=3.5.0.2 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from celery==4.0.2->-r requirements.txt (line 11)) (3.5.0.3)
Requirement not upgraded as not directly required: asn1crypto>=0.21.0 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from cryptography==1.9->-r requirements.txt (line 13)) (0.24.0)
Requirement not upgraded as not directly required: idna>=2.1 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from cryptography==1.9->-r requirements.txt (line 13)) (2.6)
Requirement not upgraded as not directly required: cffi>=1.7 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from cryptography==1.9->-r requirements.txt (line 13)) (1.11.5)
Requirement not upgraded as not directly required: sampledata in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from django-sampledatahelper==0.4.1->-r requirements.txt (line 19)) (0.3.7)
Requirement not upgraded as not directly required: MarkupSafe>=0.23 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from jinja2==2.9.6->-r requirements.txt (line 27)) (1.0)
Requirement not upgraded as not directly required: docopt>=0.5 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from psd-tools==1.4->-r requirements.txt (line 31)) (0.6.2)
Requirement not upgraded as not directly required: future in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from pyjwkest==1.3.2->-r requirements.txt (line 34)) (0.16.0)
Requirement not upgraded as not directly required: pycryptodomex in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from pyjwkest==1.3.2->-r requirements.txt (line 34)) (3.6.1)
Requirement not upgraded as not directly required: contextlib2 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from raven==6.1.0->-r requirements.txt (line 38)) (0.5.5)
Requirement not upgraded as not directly required: oauthlib>=0.6.2 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from requests-oauthlib==0.8.0->-r requirements.txt (line 40)) (2.1.0)
Requirement not upgraded as not directly required: chardet<3.1.0,>=3.0.2 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from requests==2.17.3->-r requirements.txt (line 41)) (3.0.4)
Requirement not upgraded as not directly required: urllib3<1.22,>=1.21.1 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from requests==2.17.3->-r requirements.txt (line 41)) (1.21.1)
Requirement not upgraded as not directly required: certifi>=2017.4.17 in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from requests==2.17.3->-r requirements.txt (line 41)) (2018.4.16)
Requirement not upgraded as not directly required: webencodings in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from html5lib>=0.99999999->bleach==2.0.0->-r requirements.txt (line 10)) (0.5.1)
Requirement not upgraded as not directly required: pycparser in /home/taiga/.virtualenvs/taiga/lib/python3.5/site-packages (from cffi>=1.7->cryptography==1.9->-r requirements.txt (line 13)) (2.18)
Building wheels for collected packages: Markdown, django-pglocks
  Running setup.py bdist_wheel for Markdown ... done
  Stored in directory: /tmp/pip-ephem-wheel-cache-pfs_4j1t/wheels/54/d8/4e/c42d3fac79a0e88ca7b6a2ac8a3b14f68c0a3e7b367df384e4
  Running setup.py bdist_wheel for django-pglocks ... done
  Stored in directory: /tmp/pip-ephem-wheel-cache-pfs_4j1t/wheels/fb/d0/35/3b2ebb60f1a2378b9a9bae075d436e44556cc08d9abe978858
Successfully built Markdown django-pglocks

asana 0.6.5 has requirement requests==2.14.*,>=2.14.2, but you'll have requests 2.17.3 which is incompatible.
asana 0.6.5 has requirement requests-oauthlib==0.6.*,>=0.6.1, but you'll have requests-oauthlib 0.8.0 which is incompatible.
requests 2.17.3 has requirement idna<2.6,>=2.5, but you'll have idna 2.6 which is incompatible.

Installing collected packages: Markdown, django-pglocks
  Found existing installation: Markdown 3.0.dev0
    Uninstalling Markdown-3.0.dev0:
      Successfully uninstalled Markdown-3.0.dev0
  Found existing installation: django-pglocks 1.0.2
    Uninstalling django-pglocks-1.0.2:
      Successfully uninstalled django-pglocks-1.0.2
Successfully installed Markdown-3.0.dev0 django-pglocks-1.0.2
```

**If the current behavior is a bug, please provide the steps to reproduce.**
```bash
cd ~/taiga-back
git checkout stable
workon taiga
git pull
pip install --upgrade -r requirements.txt
```

**What is the expected behavior?**
No reported errors about version incompatibilites

**Is it happening in taiga.io or in your own instance?**
Our instance

**What browser/version are you using?**
Latest Chrome

**Are there any console errors *(Ctrl + F12)* in red?**
```
GET http://taiga.cen.sor.ed/api/v1/tasks/by_ref?include_attachments=1&milestone=103&project=45&ref=19 500 (Internal Server Error)
(anonymous) @ angular.js:11756
g @ angular.js:11517
c @ angular.js:11227
a @ angular.js:15961
(anonymous) @ angular.js:15977
$eval @ angular.js:17229
$digest @ angular.js:17045
$apply @ angular.js:17337
(anonymous) @ angular.js:17633
o @ angular.js:5869
(anonymous) @ angular.js:6145
setTimeout (async)
c.defer @ angular.js:6143
k @ angular.js:17632
$applyAsync @ angular.js:17366
u @ angular.js:11569
_ @ angular.js:11778
k.onload @ angular.js:11711
load (async)
(anonymous) @ angular.js:11694
g @ angular.js:11517
c @ angular.js:11227
a @ angular.js:15961
(anonymous) @ angular.js:15977
$eval @ angular.js:17229
$digest @ angular.js:17045
$apply @ angular.js:17337
(anonymous) @ angular.js:13299
dispatch @ jquery.js:4737
g.handle @ jquery.js:4549
```

Thanks for reporting!

**Comment by Pitel** [comment]
Probably some issue with pip 10

**Comment by Pitel** [comment]
Well somehow it failed and Taiga looked pretty broken in browser. See the provided console error.

**Comment by Pitel** [comment]
Is it fixed?

**Comment by migonzalvar** [comment]
@Pitel Yes, it was fixed in 2a436d53dd8ab1bec2855202ce7188e6a8e0b07d
